components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/UserManagementAuditLogger.java

```diff
@@ -416,11 +464,31 @@ private String createAuditMessage(String action, String target, JSONObject data,
         if (data == null) {
             data = new JSONObject();
         }
+        String initiator = getInitiator();
         addContextualAuditParams(data);
         String auditMessage =
                 ListenerUtils.INITIATOR + "=%s " + ListenerUtils.ACTION + "=%s " + ListenerUtils.TARGET + "=%s "
                         + ListenerUtils.DATA + "=%s " + ListenerUtils.OUTCOME + "=%s";
-        return String.format(auditMessage, ListenerUtils.getUser(), action, target, data, resultField);
+        return String.format(auditMessage, initiator, action, target, data, resultField);
+    }
+
+    /**
+     * Get the initiator.
+     *
+     * @return the initiator based on whether log masking is enabled or not.
+     */
+    private String getInitiator() {
```

Comment:
is the same logic used in both the audit logger classes in this component? if so let's move this to a util method